sdcm/wait.py

```diff
@@ -124,3 +125,26 @@ def retry_logger(retry_state: tenacity.RetryCallState) -> None:
     )
 
     return cast(R, retry(func))
+
+
+@contextmanager
+def wait_for_log_lines(node, start_line_patterns, end_line_patterns, start_timeout=60, end_timeout=120):
+    """Waits for given lines patterns to appear in node logs despite exception raised"""
+    start_follower = node.follow_system_log(patterns=start_line_patterns)
+    end_follower = node.follow_system_log(patterns=end_line_patterns)
+    start_time = time.time()
+    try:
+        yield
+    finally:
+        started = any(start_follower)
+        while not started and (time.time() - start_time < start_timeout):
```

Comment:
I'm not sure I under stand this logic
if it's the finally clause why we need a loop like this for the start message ?

Reply:
When running `finally` quickly after command start we may still not get the starting line, so we need to wait. Besides in case this line does not show up, we may want to fail faster than waiting for full timeout.